File: 382/382895/382895_01.py
def A(n):
    m = n
    for i in map(int, str(n)):
        if i != 0 and m % i == 0:
            m //= i
    return m
def A382895(n):
    return [A(i) for i in range(1, n + 1)]

n = 10000
ary = A382895(n)
for i in range(1, n + 1):
    print(i, ary[i - 1])
    

File: 382/382895/382895_01.rb
def A(n)
  m = n
  n.to_s.split('').map(&:to_i).each{|i|
    m /= i if i != 0 && m % i == 0
  }
  m
end

def A382895(n)
  (1..n).map{|i| A(i)}
end

n = 10000
ary = A382895(n)
(1..ary.size).each{|i|
  print i
  print ' '
  puts ary[i - 1]
}